### DISCUSSION TITLE
Sofar: Proposing some renamings

### DISCUSSION THREAD

**Discussion by cschlipf** [wills106/homeassistant-solax-modbus#747] [Ideas]
I would like to propose some label renamings for the Sofar entities to increase the readability in the device view and improve clarity

|  Old name          |  New name         |  Reason         |
| --------------- | ---------------- | ------------- |
| Passive Mode Battery Charge/Discharge | Passive: Update Battery Charge/Discharge | Like for TOU, shorten the prefix to allow more space for the real entity name. Change to reflect what the button is actually doing |
| Update Charge/Discharge Times | TOU: Update Charge/Discharge Times | Group under TOU prefix |
| Update TOU Charge Times  | TOU: Update Charge Times  | Group under TOU prefix  |
| Passive Mode Grid Power  | Passive: Desired Grid Power  |  Group under PM prefix for 'passive mode'. See also #745 |
| Passive Mode Min Batter Power  |  Passive: Minimum Battery Power  | Group under PM prefix for 'passive mode'. See also #745  |
| Passive Mode Max Batter Power  | Passive: Maximum Battery Power  | Group under PM prefix for 'passive mode'. See also #745  |
| TOU ID  | TOU: ID  | Group under TOU prefix  |
| TOU Target SOC  | TOU: Target SOC  |  Group under TOU prefix  |
| TOU Charge Power | TOU: Charge Power  |  Group under TOU prefix  |
| TOU Control |  TOU: Control |  Group under TOU prefix  |
|  TOU Charge Start Time | TOU: Charge Start Time  | Group under TOU prefix  |
|  TOU Charge End Time | TOU: Charge End Time  | Group under TOU prefix  |
| RO Reflux Control  | Reflux Control | Sensor is always read-only  |
|  RO Timing ID |  Timing ID |  Sensor is always read-only  |
|  RO Timing Charge | Timing Charge  |  Sensor is always read-only  |
|  RO Timing Charge Start Time | Timing Charge Start Time  |  Sensor is always read-only |
|  RO Timing Discharge Start Time |  Timing Discharge Start Time |  Sensor is always read-only |
|  RO Timing Discharge End Time |  Timing Discharge End Time |  Sensor is always read-only |
|  RO TOU Control |  TOU: Control |  Sensor is always read-only. Group under TOU prefix. |
|  RO TOU Charge Start Time | TOU: Charge Start Time  |  Sensor is always read-only. Group under TOU prefix.  |
|  TOU Target SOC |  TOU: Target SOC |  Group under TOU prefix.  |
|  TOU Charge Power |  TOU: Charge Power |  Group under TOU prefix.  |
|  Passive Mode Timeout |  Passive: Timeout |  Group under PM prefix.  |
|  RO Passive Mode Gdes |  Passive: Desired Grid Power |  Group under PM prefix. Align with setting. See also #745  |
|  RO Passive Mode Lower |  Passive: Minimum Battery Power |  Group under PM prefix. Align with setting. See also #745  |
|  RO Passive Mode Upper |  Passive: Maximum Battery Power |  Group under PM prefix. Align with setting. See also #745  |

Let me know what you think about that. Will contribute these changes if OK.

**Comment by cschlipf** [comment]
Just to be clear: This is just about changing the labels (name). Not any of the entity IDs to not break any existing automations.

**Reply by wills106** [reply]
Seems good to me.

The only thing to be careful with in the future is that we don't abbreviate Parallel Mode to PM, like I have done in SolaX.
Otherwise we will have two groups of PM's. At the moment they all start Parallel in Sofar.

**Reply by cschlipf** [reply]
Sounds good to me. For Sofar we also have 'Timing', but I didn't change that as well and it's just one word.

Thank you very much. Expect a PR soon

**Reply by cschlipf** [reply]
Changed "PM" to "Passive" after trying it out.

**Comment by cschlipf** [comment]
PR: #748
